src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,51 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private final MachineProducer<Bulldozer> bulldozerProducer;
+    private final MachineProducer<Truck> truckProducer;
+    private final MachineProducer<Excavator> excavatorProducer;
+
+    public MachineServiceImpl() {
+        bulldozerProducer = new BulldozerProducer();
+        excavatorProducer = new ExcavatorProducer();
+        truckProducer = new TruckProducer();
+    }
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines = Collections.emptyList();
+        if (type == Bulldozer.class) {
+            machines = bulldozerProducer.get();
+        }
+        if (type == Excavator.class) {
+            machines = excavatorProducer.get();
+        }
+        if (type == Truck.class) {
+            machines = truckProducer.get();
+        }
+        return new ArrayList<>(machines);
```

Comment:
Your solution looks good but this code must be a good example of polymorphism.
Let's create an abstract variable of a type MachineProducer and initialize it with an applicable producer implementation. Then just call the method get one time.

------
(optional) Also, you can try to use a hashMap instead of if-blocks. It is a really good practice.